svc_test.go

```diff
@@ -87,6 +89,23 @@ func TestShutdown(t *testing.T) {
 	}
 }
 
+func TestContextCanceled(t *testing.T) {
+	dummyWorker := &WorkerMock{
+		InitFunc: func(*zap.Logger) error { return nil },
+		RunFunc: func() error {
+			return fmt.Errorf("stopped: %w", context.Canceled)
+		},
+		TerminateFunc: func() error { return nil },
+	}
+
+	s, err := New("dummy-service", "v0.0.1")
+	require.NoError(t, err)
+	s.AddWorker("dummy-worker", dummyWorker)
+
+	// Run should not log fatal
+	s.Run()
```

Comment:
I did some TDD while checking, so this test was failing before fixing the flow because it logged a fatal error and exited. Now it passes.